### ISSUE TITLE
Thoughts on minor version updates for when database no longer works with previous version?

### ISSUE THREAD

**Issue by rpetit3** [ncbi/amr#70]
Hi @evolarjun!

I guess this really depends on how often new databases become incompatible with previous versions. But I'm wondering if it would be useful to users if these changes came with a minor version update (e.g. 3.10.1 -> 3.11.1) instead of patches (e.g. 3.10.1 -> 3.11.2). At least this 

I think to strictly follow semantic versioning, these incompatible changes would warrant a major version update (e.g. 3.10.1 -> 4.0.0), but that's a bit much for this situation.

Just picking your brain a little to see what you think,

Thanks as always,
Robert

**Comment by rpetit3** [comment]
Hello! Me again, just saw the latest release (`3.10.20`) is no longer compatible with previous version (`3.10.18`).

How long will `3.10.18` be able to download/update databases, before `Database requires software version at least X.Y.Z` occurs?

If its ok, I'd again like you to reconsider your use of versioning. Based on the `3.10.20` release notes (major bug fix and not backwards compatible), a `3.11.1` version seemed appropriate.

Thanks as always!
Robert

**Comment by vbrover** [comment]
>  the latest release (3.10.20) is no longer compatible with previous version (3.10.18).

Compatibility is between a software version and a database version, not between a software and a software version.

> How long will 3.10.18 be able to download/update databases, before Database requires software version at least X.Y.Z occurs?

As long as the current (= latest) software version is 3.10.X.

I see your point!
**There is no change of the database format!**
There was a problem with the web page https://ftp.ncbi.nlm.nih.gov/pathogen/Antimicrobial_resistance/AMRFinderPlus/database/.

**Comment by vbrover** [comment]
And the old version 3.10.18 should work now!

**Comment by evolarjun** [comment]
Hi Robert,

For a few years now AMRFinderPlus has used HTTP to access ftp.ncbi.nlm.nih.gov when you run `amrfinder -u` instead of FTP (we're supposed to be moving away from FTP as a policy).  Because of that and the nearly direct translation from the old FTP-based version, AMRFinderPlus is dependent on the web server generated file list. Yesterday sometime the admins changed the format of the file list in a way that broke AMRFinderPlus sometime yesterday. We discovered things were broken around 5pm yesterday and Slava quickly patched the software to handle the new HTML file list format. 

After I completed the testing and release process this morning we figured out that we could write our own file list as a static file and replace the dynamically generated one, so we did that in the format that older versions of AMRFinderPlus understand.  That way the old versions still work. Longer term we don't want to be maintaining fake dynamically generated file lists, but at least for now everything is ok with any version of the software newer than mid-2019 (versions 3.2+).

Thanks for keeping an eye on it, and I'll change the release notes to be more explicit. I wrote them very quickly this morning.

Arjun
